tests/Phinx/Db/Adapter/MysqlAdapterTest.php

```diff
@@ -157,7 +157,7 @@ public function testCreateTableWithForeignKeys()
 
         $table = new \Phinx\Db\Table('ntable', array(), $this->adapter);
         $table->addColumn('realname', 'string')
-              ->addColumn('tag_id', 'integer')
+              ->addColumn('tag_id', 'integer', array('signed' => false))
```

Comment:
This is certainly a valid worry. I haven't used Phinx for any other SQL dialect but looking at e.g PostgresAdapter the behaviour there is the same in that it uses integer (signed) rather than serial (unsigned).

I am not familiar enough with Postgres to say what the common behaviour for auto-incrementing ids is, but in my experience with MySQL they are always unsigned. 

The PR is motivated by implementing Phinx in an existing large codebase / database. Applying foreign keys fails because the column types of the referencing / referenced columns do not match.  Our workaround with Phinx in production is to always specify our own primary key, and this is what brought me to write the PR.

I have to say that I had no considered portability until this point.

MySQL / Postgres could possibly be made compatible by applying a similar change and internally juggling the type of the column to Serial when it is set to unsigned? Looks like this might be the current behaviour for Identity columns in PostgresAdapter::getColumnSqlDefinition but would require some changes to the signature of getSqlType for other columns. 

I do not know enough about FKs in Postgres to comment on how they are used there, and know even less about SqlServer.